pydm/widgets/image.py

```diff
@@ -287,18 +287,39 @@ def image_width_changed(self, new_width):
             return
         self._image_width = int(new_width)
 
+    @staticmethod
+    def process_image(image):
+        """
+        Boilerplate method to be used by applications in order to add calculations and also modify the image before
+        it is displayed at the widget.
+
+        Parameters
+        ----------
+        image : np.ndarray
+            The Image Data as a 2D numpy array
+
+        Returns
+        -------
+        np.ndarray
+            The Image Data as a 2D numpy array after processing.
+        """
+        return image
+
     def redrawImage(self):
         """
         Set the image data into the ImageItem, if needed.
 
         If necessary, reshape the image to 2D first.
         """
+        self.redraw_timer.stop()
         if not self.needs_redraw:
+            self.redraw_timer.start()
             return
         image_dimensions = len(self.image_waveform.shape)
         if image_dimensions == 1:
             if self.imageWidth < 1:
                 # We don't have a width for this image yet, so we can't draw it
+                self.redraw_timer.start()
                 return
             if self.readingOrder == ReadingOrder.Clike:
```

Comment:
There is a bug here: `imageChannel` and `widthChannel` are independent, so `self.imageWidth` can be updated before some new `self.image_waveform` with this new width is received. It was a fix I was going to suggest.
In my application, where I reimplement `redrawImage`, I needed to put this `if self.readingOrder...` within a `try...except ValueError...` (where I just return to avoid breaking).

Reply:
That makes sense for all control systems in which the variables are transmitted in a separated form. 😞 I will keep that in mind for the code that I am writing now.
PS: Would you mind to share a bit more of your current use-case?

Reply:
Ok! Thank you :)
Of course I can share. It is a work in progress, it was only tested with simulations. If you would like to take a look, it is available at https://github.com/lnls-sirius/hla/blob/master/pyqt-apps/siriushla/widgets/scrn_view.py.